File: smoke_tests/ruby/hardcoded_secrets.rb
# Hardcoded Secrets vulnerabilities in Ruby

class HardcodedSecretsVulnerabilities
  # VULNERABLE: Hardcoded API key
  API_KEY = "sk_live_ruby1234567890"

  # VULNERABLE: Hardcoded password
  DB_PASSWORD = "super_secret_password"

  def initialize
    # VULNERABLE: Hardcoded credentials
    @aws_access_key = "AKIAIOSFODNN7EXAMPLE"
    @aws_secret_key = "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
  end

  def connection_string
    # VULNERABLE: Hardcoded database credentials
    "postgres://admin:password123@localhost:5432/myapp"
  end

  def jwt_secret
    # VULNERABLE: Hardcoded JWT secret
    "my_super_secret_jwt_key_ruby"
  end

  def authenticate(username, password)
    # VULNERABLE: Hardcoded backdoor
    return true if password == "backdoor_ruby_123"
    false
  end

  def encryption_key
    # VULNERABLE: Hardcoded encryption key
    "\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0a\x0b\x0c\x0d\x0e\x0f\x10"
  end
end

File: smoke_tests/php/hardcoded_secrets.php
<?php
// Hardcoded Secrets vulnerabilities in PHP

// VULNERABLE: Hardcoded API key
define('API_KEY', 'sk_live_php1234567890');

// VULNERABLE: Hardcoded password
const DB_PASSWORD = 'super_secret_password';

class HardcodedSecretsVulnerabilities {
    // VULNERABLE: Hardcoded AWS credentials
    private $awsAccessKey = 'AKIAIOSFODNN7EXAMPLE';
    private $awsSecretKey = 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY';

    public function getConnectionString() {
        // VULNERABLE: Hardcoded connection string
        return 'mysql:host=localhost;dbname=myapp;user=admin;password=admin123';
    }

    public function getJwtSecret() {
        // VULNERABLE: Hardcoded JWT secret
        return 'my_super_secret_jwt_key_php';
    }

    public function authenticate($username, $password) {
        // VULNERABLE: Hardcoded backdoor
        if ($password === 'backdoor_php_123') {
            return true;
        }
        return false;
    }

    public function getEncryptionKey() {
        // VULNERABLE: Hardcoded encryption key
        return '0123456789abcdef';
    }

    public function connectDatabase() {
        // VULNERABLE: Hardcoded credentials
        $host = 'localhost';
        $user = 'root';
        $pass = 'mysql_password_123';
        return mysqli_connect($host, $user, $pass);
    }
}
